clients/spark/core/src/main/scala/io/treeverse/clients/GarbageCollector.scala

```diff
@@ -475,23 +517,55 @@ object GarbageCollector {
     s"$gcAddressesLocation/$markId.meta"
   }
 
-  private def generateMarkMetadataDataframe(runId: String): DataFrame = {
-    val metadata = List(runId).map(Tuple1(_))
-    val fields = Array(StructField(RUN_ID_KEY, StringType, nullable = false))
+  private def getMarkMetadata(markId: String, gcAddressesLocation: String): DataFrame = {
+    val addressesMarkMetadataLocation = getMetadataMarkLocation(markId, gcAddressesLocation)
+    spark.read.json(addressesMarkMetadataLocation)
+  }
+
+  private def populateRunIDAndCommitsLocation(
+      markID: String,
+      gcAddressesLocation: String
+  ): Array[String] = {
+    import spark.implicits._
+    val markMetadataDF = getMarkMetadata(markID, gcAddressesLocation)
+    markMetadataDF
+      .select(RUN_ID_KEY, COMMITS_LOCATION_KEY)
+      .map(row => {
+        var arr = Array[String]()
+        for (i <- 0 until row.length) {
+          arr = arr :+ row.getString(i)
+        }
+        arr
+      })
+      .first()
```

Comment:
What does this part do?

Reply:
It transforms the first Row of the DataFrame into Array[String]